siad/daemon.go

```diff
@@ -57,6 +57,10 @@ func processModules(modules string) (string, error) {
 func processConfig(config Config) (Config, error) {
 	var err error
 	config.Siad.APIaddr = processNetAddr(config.Siad.APIaddr)
+	apiaddr := modules.NetAddress(config.Siad.APIaddr)
+	if apiaddr.Host() != "" && !apiaddr.IsLoopback() && !config.Siad.AllowAPIBind {
+		return Config{}, errors.New("ERROR: You must pass --disable-security to bind Siad to a non-localhost address.")
```

Comment:
I would structure it like this:
```go
if !config.Siad.AllowAPIBind {
    addr := modules.NetAddress(config.Siad.APIaddr)
    if !addr.IsLoopback() && addr.Host() != "" {
        return Config{}, errors.New("you must pass --disable-security to bind Siad to a non-localhost address")
    }
}
```
Although the `.Host() != ""` really shouldn't be necessary; that seems like a failing of `IsLoopback`.